api/src/permissions/modules/process-payload/process-payload.ts

```diff
@@ -94,15 +99,16 @@ export async function processPayload(options: ProcessPayloadOptions, context: Co
 			context,
 		);
 
+		// only query fields not in the payload
+		if (!(field.field in payloadWithPresets)) {
+			fieldsToQuery.push(field.field);
```

Comment:
Ideally we only add select fields that are not in the payload + present in the validation rules. Due to the recursive nature of the possible rules I am not convinced it is more performant to do so over the basic check to ensure it is not in the payload. Thoughts?